Kernel/Locking/RAII.cpp

```diff
@@ -0,0 +1,22 @@
+/**
+ * @file RAII.cpp
+ * @author Keeton Feavel (keeton@xyr.is)
+ * @brief
+ * @version 0.1
+ * @date 2022-01-26
+ *
+ * @copyright Copyright the Xyris Contributors (c) 2022
+ *
+ */
+#include "RAII.hpp"
+
+RAIIMutex::RAIIMutex(Mutex& mutex)
```

Comment:
I really feel like this code should be in the header file. Otherwise it's going to add extra overhead.